src/cpp/database/entities.hpp

```diff
@@ -0,0 +1,363 @@
+/* Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file entities.hpp
+ */
+
+#ifndef _EPROSIMA_FASTDDS_STATISTICS_BACKEND_DATABASE_ENTITIES_HPP_
+#define _EPROSIMA_FASTDDS_STATISTICS_BACKEND_DATABASE_ENTITIES_HPP_
+
+#include "data.hpp"
+#include <fastdds-statistics-backend/types/types.hpp>
+#include <fastdds-statistics-backend/types/EntityId.hpp>
+
+#include <string>
+
+namespace eprosima {
+namespace statistics_backend {
+namespace database {
+
+struct User;
+struct Process;
+struct DomainParticipant;
+struct Topic;
+struct DataReader;
+struct DataWriter;
+struct Locator;
+
+/*
+ * Base struct for all the database possible entities.
+ *
+ * Entities constitute the nodes of the database. They hold the relation they have
+ * both upwards and downwards, plus they hold the statistical data reported by Fast
+ * DDS Statistics Module.
+ */
+struct Entity
+{
+    Entity(
+            EntityKind entity_kind = EntityKind::INVALID,
+            std::string entity_name = "INVALID")
+        : kind(entity_kind)
+        , name(entity_name)
+    {
+    }
+
+    //! The unique identification of the entity
+    EntityId id;
+
+    //! The kind of entity
+    EntityKind kind;
+
+    //! A user defined human-readable name for the entity
+    std::string name;
+};
+
+/*
+ * Host entities hold data about the real host involved in the communication
+ */
+struct Host : Entity
+{
+    Host(
+            std::string host_name)
+        : Entity(EntityKind::HOST, host_name)
+    {
+    }
```

Comment:
And with the suggested Entity constructor, modify the subclasses like:
```suggestion
    Host(
            std::string host_name)
        : kind(EntityKind::HOST)
        , name(host_name)
{
}
```

Reply:
See above